ProjFS.Mac/PrjFSLib.Mac.Managed/VirtualizationInstance.cs

```diff
@@ -18,8 +33,12 @@ public class VirtualizationInstance
             {
                 OnEnumerateDirectory = this.OnEnumerateDirectory,
                 OnGetFileStream = this.OnGetFileStream,
+
+                // Note: We must use this.notifyOperationCallback and not this.OnNotifyOperation
+                // to ensure we don't create a temporary delegate that will be garbage collected
```

Comment:
It would be nice to make this more explicit in the code itself. Perhaps something like this?

```
OnNotifyOperation = (this.preventGCOnNotifyOperationDelegate = new NotifyOperationCallback(this.OnNotifyOperation)),
```

The root cause of the issue is that there is an implicit delegate allocation, so let's make at least that much of it explicit in the code. I think it would also be clearer to store that instance here, rather than in the constructor.

Reply:
@sanoursa 

> I think it would also be clearer to store that instance here, rather than in the constructor.

That sounds good to me (note that `preventGCOnNotifyOperationDelegate` will no longer be able to be `readonly` with this change).